pfs_middleware/pfs_middleware/middleware.py

```diff
@@ -725,8 +730,10 @@ def __call__(self, req):
                 # Check whether we ought to bypass. Note that swift_owner
                 # won't be set until we call authorize
                 if config_true_value(req.headers.get('X-Bypass-ProxyFS')) and \
+                        self.bypass_mode in ('read-only', 'read-write') and \
                         req.environ.get('swift_owner'):
-                    if method not in ('GET', 'HEAD'):
+                    if self.bypass_mode == 'read-only' and method not in (
+                            'GET', 'HEAD'):
```

Comment:
Hmm.  I think we should disallow DELETE even in the read-write case.  That way if a ProxyFS object turns up missing we don't have to worry that a "client" might have deleted it using this interface.

that does create an issue if a client creates an object but decides, for whatever reason, that it doesn't want to register the object with Proxyfs, it will not be able to delete it.

Can we do anything to prohibit the over-write of existing objects via this interface?  An object should never be updated once its created.